test/e2e/handler/rollback_test.go

```diff
@@ -75,17 +137,77 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			resetDesiredStateForNodes()
 		})
 		It("[test_id:3793]should rollback to a good gw configuration", func() {
+			By("Should not be available") // Fail fast
+			policyConditionsStatusConsistently().ShouldNot(containPolicyAvailable())
+
 			By("Wait for reconcile to fail")
 			waitForDegradedTestPolicy()
 			By(fmt.Sprintf("Check that %s is rolled back", primaryNic))
 			Eventually(func() bool {
-				return dhcpFlag(nodes[0], primaryNic)
+				return autoDNS(nodes[0], primaryNic)
 			}, 480*time.Second, ReadInterval).Should(BeTrue(), "DHCP flag hasn't rollback to true")
 
 			By(fmt.Sprintf("Check that %s continue with rolled back state", primaryNic))
 			Consistently(func() bool {
-				return dhcpFlag(nodes[0], primaryNic)
+				return autoDNS(nodes[0], primaryNic)
 			}, 5*time.Second, 1*time.Second).Should(BeTrue(), "DHCP flag has change to false")
 		})
 	})
+
+	Context("when name servers are lost after state configuration", func() {
+		BeforeEach(func() {
+			updateDesiredStateAtNode(nodes[0], removeNameServers(primaryNic))
+		})
+		AfterEach(func() {
+			updateDesiredStateAtNode(nodes[0], discoverNameServers(primaryNic))
+			By("Clean up desired state")
+			resetDesiredStateForNodes()
+		})
+		It("should rollback to previous name servers", func() {
+			By("Should not be available") // Fail fast
+			policyConditionsStatusConsistently().ShouldNot(containPolicyAvailable())
+
+			By("Wait for reconcile to fail")
+			waitForDegradedTestPolicy()
+			By(fmt.Sprintf("Check that %s is rolled back", primaryNic))
+			Eventually(func() bool {
+				return autoDNS(nodes[0], primaryNic)
+			}, 480*time.Second, ReadInterval).Should(BeTrue(), "auto-dns flag hasn't rollback to true")
+
+			By(fmt.Sprintf("Check that %s continue with rolled back state", primaryNic))
+			Consistently(func() bool {
+				return autoDNS(nodes[0], primaryNic)
+			}, 5*time.Second, 1*time.Second).Should(BeTrue(), "auto-dns flag has change to false")
+
+		})
+	})
+
+	Context("when name servers are wrong after state configuration", func() {
+		BeforeEach(func() {
+			updateDesiredStateAtNode(nodes[0], setBadNameServers(primaryNic))
+		})
+		AfterEach(func() {
+			updateDesiredStateAtNode(nodes[0], discoverNameServers(primaryNic))
+			By("Clean up desired state")
+			resetDesiredStateForNodes()
+		})
+		It("should rollback to previous name servers", func() {
+			By("Should not be available") // Fail fast
+			policyConditionsStatusConsistently().ShouldNot(containPolicyAvailable())
+
+			By("Wait for reconcile to fail")
+			waitForDegradedTestPolicy()
+			By(fmt.Sprintf("Check that %s is rolled back", primaryNic))
+			Eventually(func() bool {
+				return dhcpFlag(nodes[0], primaryNic)
```

Comment:
The title speaks about name servers, but here you check DHCP fla

Reply:
Fixed